rules/order/src/Rector/ClassMethod/OrderConstructorDependenciesByTypeAlphabeticallyRector.php

```diff
@@ -0,0 +1,135 @@
+use Rector\Core\Rector\AbstractRector;
+use Rector\Core\RectorDefinition\CodeSample;
+use Rector\Core\RectorDefinition\RectorDefinition;
+
+/**
+ * @see \Rector\Order\Tests\Rector\ClassMethod\OrderConstructorDependenciesByTypeAlphabeticallyRector\OrderConstructorDependenciesByTypeAlphabeticallyRectorTest
+ */
+final class OrderConstructorDependenciesByTypeAlphabeticallyRector extends AbstractRector
+{
+    public function getDefinition(): RectorDefinition
+    {
+        return new RectorDefinition('Order __constructor dependencies by type A-Z', [
+            new CodeSample(
+                <<<'PHP'
+class SomeClass
+{
+    public function __construct(
+        LatteToTwigConverter $latteToTwigConverter,
+        SymfonyStyle $symfonyStyle,
+        LatteAndTwigFinder $latteAndTwigFinder,
+        SmartFileSystem $smartFileSystem,
+    ) {
+    }
+}
+PHP
+,
+                <<<'PHP'
+class SomeClass
+{
+    public function __construct(
+        LatteAndTwigFinder $latteAndTwigFinder,
+        LatteToTwigConverter $latteToTwigConverter,
+        SmartFileSystem $smartFileSystem,
+        SymfonyStyle $symfonyStyle,
+    ) {
+    }
+}
+PHP
+
+            ),
+        ]);
+    }
+
+    /**
+     * @return string[]
+     */
+    public function getNodeTypes(): array
+    {
+        return [ClassMethod::class];
+    }
+
+    /**
+     * @param ClassMethod $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        if ($this->shouldSkip($node)) {
+            return null;
+        }
+
+        $node->params = $this->getSortedParams($node);
+
+        return $node;
+    }
+
+    /**
+     * @return Param[]
+     */
+    private function getSortedParams(ClassMethod $node): array
+    {
+        $params = $node->getParams();
+        usort($params, function (Param $a, Param $b) {
+            /** @var Name $typeA */
+            $typeA = $a->type;
+            /** @var Name $typeB */
+            $typeB = $b->type;
+            return strcmp($this->getShortName($typeA), $this->getShortName($typeB));
+        });
+
+        return $params;
+    }
+
+    private function isPrimitiveDataTypeParam(Param $param): bool
+    {
+        if ($param->type instanceof Identifier) {
+            return true;
+        }
+
+        return false;
+    }
+
+    private function isDefaultValueParam(Param $param): bool
+    {
+        if ($param->default instanceof Expr) {
+            return true;
+        }
+
+        return false;
+    }
+
+    private function shouldSkip(ClassMethod $node): bool
+    {
+        if ($this->getName($node) !== '__construct') {
+            return true;
+        }
+
+        if ($node->params === []) {
+            return true;
+        }
+
+        foreach ($node->params as $param) {
+            if ($this->isDefaultValueParam($param)) {
+                return true;
+            }
+
+            if ($this->isPrimitiveDataTypeParam($param)) {
+                return true;
+            }
+        }
```

Comment:
@TomasVotruba Originally, I made two methods `hasDefaultValueParam` and `hasPrimitiveDataTypeParam` and they contained their own foreach loop but then I decided to have one loop. Please let me know which approach is better.

Reply:
I think both are fine :+1: